snuba/query/snql/parser.py

```diff
@@ -520,23 +606,49 @@ def generic_visit(self, node: Node, visited_children: Any) -> Any:
         return generic_visit(node, visited_children)
 
 
-def parse_snql_query_initial(body: str) -> Query:
+def parse_snql_query_initial(body: str) -> LogicalQuery:
     """
     Parses the query body generating the AST. This only takes into
     account the initial query body. Extensions are parsed by extension
     processors and are supposed to update the AST.
     """
     exp_tree = snql_grammar.parse(body)
     parsed = SnQLVisitor().visit(exp_tree)
-    assert isinstance(parsed, Query)  # mypy
+    assert isinstance(parsed, LogicalQuery)  # mypy
     return parsed
 
 
-def parse_snql_query(body: str, dataset: Dataset) -> Query:
+DATETIME_MATCH = FunctionCallMatch(
+    StringMatch("toDateTime"), (Param("date_string", LiteralMatch(AnyMatch(str))),)
+)
+
+
+def _parse_datetime_literals(query: LogicalQuery) -> None:
+    def parse(exp: Expression) -> Expression:
+        result = DATETIME_MATCH.match(exp)
+        if result is not None:
+            date_string = result.expression("date_string")
+            assert isinstance(date_string, Literal)  # mypy
+            assert isinstance(date_string.value, str)  # mypy
+            return Literal(exp.alias, parse_datetime(date_string.value))
+
+        return exp
+
+    query.transform_expressions(parse)
+
+
+def parse_snql_query(body: str, dataset: Dataset) -> LogicalQuery:
     query = parse_snql_query_initial(body)
-    _validate_aliases(query)
-    _parse_subscriptables(query)
-    _apply_column_aliases(query)
-    _expand_aliases(query)
-    _deescape_aliases(query)
+
+    # These are the post processing phases
+    _parse_datetime_literals(query)
+
+    # TODO: Update these functions to work with snql queries in a separate PR
+    _validate_aliases(query)  # -> needs to recurse through sub queries
+    _parse_subscriptables(query)  # -> This should be part of the grammar
+    _apply_column_aliases(query)  # -> needs to recurse through sub queries
+    _expand_aliases(query)  # -> needs to recurse through sub queries
+    _deescape_aliases(
```

Comment:
You need to bring over the alias mangling function.
https://github.com/getsentry/snuba/blob/master/snuba/query/parser/__init__.py#L386

Reply:
I will in a subsequent PR.